File: backend/Models/Issue.cs
namespace backend.Models
{
    public class Issue
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;
        
        // Input Factors
        public string Urgency { get; set; } = "LOW";
        public int AffectedPeople { get; set; } = 1;
        
        // Calculated Priority
        public int PriorityScore { get; set; }
        public string PriorityLevel { get; set; } = "LOW";
        
        // Stored as JSON or separated string in DB
        public string PriorityFactors { get; set; } = string.Empty;
        
        // State
        public string Status { get; set; } = "PENDING";
        
        // Relations
        public int ReportedById { get; set; }
        public User? ReportedBy { get; set; }
        
        public int? AssignedToId { get; set; }
        public User? AssignedTo { get; set; }
        
        // Timestamps
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ResolvedAt { get; set; }
    }
}

File: frontend/src/types/issue.ts
export type IssueStatus = 'PENDING' | 'ASSIGNED' | 'IN_PROGRESS' | 'RESOLVED' | 'VERIFIED';
export type PriorityLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
export type UrgencyLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export interface EvidenceItem {
  id: number;
  imageType: 'PROBLEM' | 'RESOLUTION';
  url: string;
  uploadedByName: string;
  createdAt: string;
}

export interface Issue {
  id: number;
  title: string;
  description: string;
  category: string;
  location: string;
  urgency: UrgencyLevel;
  affectedPeople: number;
  priorityScore: number;
  priorityLevel: PriorityLevel;
  priorityFactors: string[];
  status: IssueStatus;
  reportedById: number;
  reportedByName: string;
  assignedToId?: number;
  assignedToName?: string;
  createdAt: string;
  updatedAt: string;
  resolvedAt?: string;
  // Verification
  verifiedByName?: string;
  verifiedAt?: string;
  reworkNotes?: string;
  // Evidence
  problemEvidence: EvidenceItem[];
  resolutionEvidence: EvidenceItem[];
}

export interface CreateIssueRequest {
  title: string;
  description: string;
  category: string;
  location: string;
  urgency: UrgencyLevel;
  affectedPeople: number;
}

export interface CreateIssueResponse {
  issue: Issue;
  possibleDuplicate: boolean;
  existingIssueId?: number;
  similarityReason?: string;
}

export interface UpdateIssueRequest {
  status?: IssueStatus;
  assignedTo?: number;
  priorityLevel?: PriorityLevel;
}

export interface Comment {
  id: number;
  issueId: number;
  userId: number;
  userName: string;
  userRole: string;
  content: string;
  isWorkNote: boolean;
  createdAt: string;
}

export interface PriorityResult {
  priorityScore: number;
  priorityLevel: PriorityLevel;
  factors: string[];
}
